src/Hl7.Fhir.Specification/Validation/Validator.cs

```diff
@@ -76,66 +77,72 @@ internal Validator NewInstance()
             {
                 OnSnapshotNeeded = this.OnSnapshotNeeded,
                 OnExternalResolutionNeeded = this.OnExternalResolutionNeeded,
+                modelInspector = this.modelInspector,
 
 #if REUSE_SNAPSHOT_GENERATOR
                 _snapshotGenerator = this._snapshotGenerator
 #endif
             };
         }
 
-        public OperationOutcome Validate(ITypedElement instance)
+        public OperationOutcome Validate(ITypedElement instance, ModelInspector inspector)
```

Comment:
This is probably our biggest change - and I am not sure we can make it. If you operating in a fully ITypedElement world, you didn't need a ModelInspector until now.  And in fact, for FS and probably Simplifier - this is going to be a problem. They do not have a ModelInspector.  They have a StructureDefinitionSummaryProvider.  I guess we didn't need this before because the validator was not common - so it could use "IsSubtypeFor" etc without a modelinspector.

Now, normally, this would be a show stopper - but we are also going to replace the validator, and make this a shim. I am guessing we don't need the ModelInspector anymore (since our new validator is not model-specific anymore), so this will disappear.  Well, not completely of course, since the *compiler* is still FHIR-version bound.  I think we have some challenges ahead here.

Reply:
We are aware of this and we know that this lives in the SDK for a very short time. It means that the new validator should be merged in the SDK for version 5 as well.